external/fv3fit/fv3fit/sklearn/_ocsvm_novelty_detector.py

```diff
@@ -0,0 +1,197 @@
+        )
+        model.pipeline = make_pipeline(scaler, ocsvm)
+        model.pipeline.fit(X)
+
+        seconds = time.time() - start_time
+        model.logger.info(f"OCSVM novelty detector done fitting in {seconds}s.")
+
+        model._compute_max_score_train(X)
+        model.is_trained = True
+        return model
+
+    def _compute_max_score_train(self, X):
+        self.logger.info(f"Computing maximum scores on training data.")
+        start_time = time.time()
+
+        # We negate the scores to ensure that larger scores correspond to a higher
+        # likelihood of being an outlier
+        scores = -1 * self.pipeline.score_samples(X)
+        max_score = np.max(scores)
+        self.maximum_training_score = float(max_score)
+
+        seconds = time.time() - start_time
+        self.logger.info(f"Computed largest score of {max_score} in {seconds}s.")
+
+    def _get_default_cutoff(self):
+        return self.maximum_training_score
+
+    def predict(self, data: xr.Dataset) -> xr.Dataset:
+        assert self.is_trained
+
+        start_time = time.time()
+        self.logger.info(f"Predicting with OCSVM novelty detector.")
+
+        stack_dims = [dim for dim in data.dims if dim not in stacking.Z_DIM_NAMES]
+        stacked_data = data.stack({SAMPLE_DIM_NAME: stack_dims})
+        stacked_data = stacked_data.transpose(SAMPLE_DIM_NAME, ...)
+
+        X, _ = pack(
+            stacked_data[self.input_variables], [SAMPLE_DIM_NAME], self.packer_config
+        )
+        stacked_scores = -1 * self.pipeline.score_samples(X)
+
+        new_coords = {
+            k: v
+            for (k, v) in stacked_data.coords.items()
+            if k not in stacking.Z_DIM_NAMES
+        }
+        stacked_scores = xr.DataArray(
+            stacked_scores, dims=[SAMPLE_DIM_NAME], coords=new_coords
+        )
+        score_dataset = stacked_scores.to_dataset(name=self._SCORE_OUTPUT_VAR).unstack(
+            SAMPLE_DIM_NAME
+        )
+
+        seconds = time.time() - start_time
+        self.logger.info(
+            f"Finished predicting with OCSVM novelty detector in {seconds}s."
+        )
+
+        return stacking.match_prediction_to_input_coords(data, score_dataset)
+
+    def dump(self, path: str) -> None:
+        assert self.is_trained
+
+        fs: fsspec.AbstractFileSystem = fsspec.get_fs_token_paths(path)[0]
+        fs.makedirs(path, exist_ok=True)
+
+        mapper = fs.get_mapper(path)
+
+        f = io.BytesIO()
+        joblib.dump({"pipeline": self.pipeline}, f)
+        mapper[self._PICKLE_NAME] = f.getvalue()
+
+        metadata = {
+            "input_variables": self.input_variables,
+            "packer_config": dataclasses.asdict(self.packer_config),
+            "input_features_": dataclasses.asdict(self.input_features_),
+            "gamma": self.gamma,
+            "nu": self.nu,
+            "max_iter": self.max_iter,
+            "maximum_training_score": self.maximum_training_score,
+        }
```

Comment:
Can just dump the things required for making prediction, since model has to be already trained to dump. 

```suggestion
        metadata = {
            "input_variables": self.input_variables,
            "packer_config": dataclasses.asdict(self.packer_config),
            "input_features_": dataclasses.asdict(self.input_features_),
            "maximum_training_score": self.maximum_training_score,
        }
```